feat(tem): smtp config (#3142)
scaleway/docs-content: managed-services/transactional-email/reference-content/smtp-configuration.mdx

---
meta:
  title: Setting up SMTP
  description: This page goes into detail about the different elements in a standard SMTP configuration, and how to fill out each entry to ensure an optimal configuration of Transactional Email services.
content:
  h1: Setting up SMTP
  paragraph: This page goes into detail about the different elements in a standard SMTP configuration, and how to fill out each entry to ensure an optimal configuration of Transactional Email services.
tags: smtp configuration forwarding port
dates:
  validation: 2024-05-02
  posted: 2024-05-02
categories:
  - managed-services
---

> You can set up Scaleway Transaction Email for your applications using either the [Scaleway API](https://www.scaleway.com/en/developers/api/transactional-email/) or by using the SMTP configuration options of your CRM or application management tool.

Reviewer: ```suggestion
You can set up Scaleway Transaction Email for your applications using either the [Scaleway API](https://www.scaleway.com/en/developers/api/transactional-email/) or the SMTP configuration options of your CRM or application management tool.
```

> On this page we go through different elements you might find in a standard SMTP configuration, and how to fill out each entry to ensure an optimal configuration of your Transactional Email service.

Reviewer: ```suggestion
On this page, we go through different elements you might find in a standard SMTP configuration, and how to fill out each entry to ensure an optimal configuration of your Transactional Email service.
```

> ## Standard Settings

Reviewer: ```suggestion
## Standard settings
```

1 - **SMTP Server Address (Host)** - The domain name or IP address of the SMTP server. If you are using Scaleway Transactional Email, the domain to use is `smtp.tem.scw.cloud`.

2 - **SMTP Port** - The port number used to connect to the SMTP server. At Scaleway, the default ports are `25`, `587`, and `2587`.

  Depending on your use case, we recommend using the following ports:

   - **Port 587**: Used to securely submit emails over TLS.

>    - **Port 465**: Used for SMTP connections over SSL/TLS. At Scaleway you can also use port `2465`.

Reviewer: ```suggestion
   - **Port 465**: Used for SMTP connections over SSL/TLS. At Scaleway, you can also use port `2465`.
```

   - **Port 25**: This port was traditionally used for relaying and submitting emails, but it is not recommended when sending emails directly from an application, due to widespread blocking to prevent spam.

> 3 - **Username and Password** - Sometimes you need to authenticate to the SMTP server to connect. You might be asked to provide a valid username and password.

Reviewer: ```suggestion
3 - **Username and password** - Sometimes you need to authenticate to the SMTP server to connect. You might be asked to provide a valid username and password.
```

Your Scaleway SMTP username is the Project ID of the Project in which the TEM domain was created. You can find it under **Managed Services** > **Transactional Email** > **Domain Overview** in the Scaleway console.

Your password is the secret key of the API key of the project used to manage your TEM domain. Follow this procedure [to generate API keys for API and SMTP sending with IAM](https://www.scaleway.com/en/docs/managed-services/transactional-email/how-to/generate-api-keys-for-tem-with-iam/).

> 4 - **Encryption Method** - If you want to Encrypt the connection between your application and the SMTP server, there are usually two methods available:

Reviewer: ```suggestion
4 - **Encryption method** - If you want to encrypt the connection between your application and the SMTP server, there are usually two methods available:
```

   - **SSL/TLS**: Also known as SMTPS, it allows you to directly define a secure connection on a secure port. Directly creates a secure connection on a port such as `465` and `2465`.
   - **STARTTLS**: This type will upgrade any insecure connections to secure connections on a non-secure port, such as 587.

> 5 - **Sender Email Address** - The email address that will appear as the sender. Whenever someone receives an email from your application this is the email they see.

Reviewer: ```suggestion
5 - **Sender email address** - The email address that will appear as the sender. Whenever someone receives an email from your application, this is the email they see.
```

> 6 - **Authentication Method** - This is the method you can use to ensure that only authorized users can send emails from the SMTP server, which helps avoid spam.

Reviewer: ```suggestion
6 - **Authentication method** - This is the method you can use to ensure that only authorized users can send emails from the SMTP server, which helps avoid spam.
```

SMTP supports different methods:

>     - **PLAIN**: Username and password are transmitted in plain text and secured via SSL/TLS

Reviewer 2: ```suggestion
    - **PLAIN**: Username and password are transmitted in plain text and secured via SSL/TLS.
```

>     - **LOGIN**: Username and passowrd are encoded in base64 and then transmitted through SSL/TLS.

Reviewer 2: ```suggestion
    - **LOGIN**: Username and password are encoded in base64 and then transmitted through SSL/TLS.
```

>     - **CRAM-MD5**: The most secure method where a challenge-response mechanism is used, to avoid sending the password via a network.

Reviewer 3: We are not supporting yet CRAM-MD5 login mechanism.

Note that the LOGIN method is considered as Obsolete by the iana https://www.iana.org/assignments/sasl-mechanisms/sasl-mechanisms.xhtml

>     - **PLAIN**: Username and password are transmitted in plain text and secured via SSL/TLS.
>     - **LOGIN**: Username and password are encoded in base64 and then transmitted through SSL/TLS.
>     - **CRAM-MD5**: The most secure method where a challenge-response mechanism is used, to avoid sending the password via a network.

Author: ```suggestion
    - **PLAIN**: Username and password are transmitted in plain text and secured via SSL/TLS.
    - **LOGIN**: Username and password are encoded in base64 and then transmitted through SSL/TLS. Note that this method is [considered obsolete by the Internet Assigned Numbers Authority (IANA)](https://www.iana.org/assignments/sasl-mechanisms/sasl-mechanisms.xhtml)
```

## Optional Settings

Some SMTP configurations might require additional information, such as:

- **Timeouts**: The definition of long the server is expected to wait for a response or connection before quitting the operation.
- **Connection retries**: The definition of how many times the server should try to connect if the connection fails the first time.

> - **DKIM Signing**: DomainKeys Identified Mail (DKIM) signatures allow you to check that the message was not altered in transit,

Reviewer 2: ```suggestion
- **DKIM Signing**: DomainKeys Identified Mail (DKIM) signatures allow you to check that the message was not altered in transit.
```

- **IP Address Binding**: If your server has more than one IP address, you can specify which one will be used for outgoing emails.

> ## Example Configuration

Reviewer: ```suggestion
## Example configuration
```
